code/clue.js

```diff
@@ -1,59 +1,495 @@
-  weight: 10
-}
+  id: "rope",
+  name: "Rope",
+  weight: 10,
+  resident: "Hall",
+  image: "img/rope-1295003_640.png",
+};
+
+const knife = {
+  id: "knife",
+  name: "Knife",
+  weight: 1,
+  resident: "Kitchen",
+  image: "img/knife-2930492_640.png",
+};
+
+const candlestick = {
+  id: "candlestick",
+  name: "Candlestick",
+  weight: 6,
+  resident: "Ball room",
+  image: "img/candles-3913775_640.png",
+};
 
-// THE ROOMS ONLY HAS A NAME SO NO NEED FOR OBJECTS THERE.
+const dumbbell = {
+  id: "dumbbell",
+  name: "Dumbbell",
+  weight: 20,
+  resident: "Spa",
+  image: "img/black-1295124_640.png",
+};
 
-// NOW GROUP ALL SUSPECTS, WEAPONS AND ROOMS IN ARRAYS LIKE THIS:
+const poison = {
+  id: "poison",
+  name: "Poison",
+  weight: 2,
+  resident: "Conservatory",
+  image: "img/poison-576608_640.png",
+};
+
+const axe = {
+  id: "axe",
+  name: "Axe",
+  weight: 50,
+  resident: "Patio",
+  image: "img/axe-159659_640.png",
+};
+
+const bat = {
+  id: "bat",
+  name: "Bat",
+  weight: 1.5,
+  resident: "Billiard room",
+  image: "img/base-25755_640.png",
+};
+
+const trophy = {
+  id: "trophy",
+  name: "Trophy",
+  weight: 10,
+  resident: "Lounge",
+  image: "img/cup-1757500_640.png",
+};
+
+const pistol = {
+  id: "pistol",
+  name: "Pistol",
+  weight: 2,
+  resident: "Theater",
+  image: "img/gun-5517424_640.png",
+};
+
+// Groups for all the suspects, weapons and rooms
 
 const suspects = [
   mrGreen,
-  mrsWhite
-  // ...  and the rest
-]
+  professorPlum,
+  missScarlet,
+  mrsPeacock,
+  colonelMustard,
+  mrsWhite,
+];
+
+const weapons = [
+  rope,
+  knife,
+  candlestick,
+  dumbbell,
+  poison,
+  axe,
+  bat,
+  trophy,
+  pistol,
+];
+
+const rooms = [
+  "Dining Room",
+  "Conservatory",
+  "Kitchen",
+  "Study",
+  "Library",
+  "Billiard Room",
+  "Lounge",
+  "Ballroom",
+  "Hall",
+  "Spa",
+  "Living Room",
+  "Observatory",
+  "Theater",
+  "Guest House",
+  "Patio",
+];
 
-const weapons = []
+// Empty object to store the randomly picked killer, weapon and room.
+let mystery = {
+  killer: "",
+  weapon: "",
+  room: "",
+};
 
-const rooms = []
+// Randomly select an item from an array
+const randomSelector = (array) => {
+  return array[Math.floor(Math.random() * array.length)];
+};
 
-// THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
-// YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
-const randomSelector = array => {
-  return array[Math.floor(Math.random() * array.length)]
-}
+// Fill an array with similar objects
+const fillArray = (item, amountOfTimes) => {
+  let result = [];
+  for (let i = 0; i < amountOfTimes; i++) {
+    result.push(item);
+  }
+  return result;
+};
 
-// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
+// Create new weapon array if suspect has a favourite weapon
+const favouriteWeaponArray = (weapons) => {
+  if (mystery.killer.favouriteWeapon) {
```

Comment:
Great job with how you write your statements!
For further reference, check out https://www.freecodecamp.org/news/js-type-coercion-explained-27ba3d9a2839/